### PR TITLE
Update Rm API endpoint to work with both not opted-in and opted-in opertors

### PR DESCRIPTION
**Pull request by lipoja**

### CODE REVIEW

**Review by yashvardhannanavati — COMMENTED** [review]
just one comment, otherwise looks good

**Comment by chandwanitulsi** [comment]
Code LGTM, but while testing its failed on `IIBError('config directory does not exist: /tmp/iib-emgeriaw/from_index/configs')`, maybe it just needs makedir here as well
Building dev-env again to test

**Comment by yashvardhannanavati** [comment]
Hi Jan, when I run this on my local, 

`curl -u: --negotiate -X POST -H "Content-Type: application/json" http://localhost:8080/api/v1/builds/rm -d '{"operators":["submariner"],"from_index":"registry-proxy.engineering.redhat.com/rh-osbs/iib-pub-pending:v4.11", "binary_image": "registry.redhat.io/openshift4/ose-operator-registry:v4.11"}'`

fails with


`[2023-02-07 22:39:22,972: ERROR/ForkPoolWorker-1] Task iib.workers.tasks.build.handle_rm_request[72b7ee73-2925-4c17-ada6-55260c971931] raised unexpected: IIBError('config directory does not exist: /tmp/iib-u6rnxy3l/from_index/configs')
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/celery/app/trace.py", line 451, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/celery/app/trace.py", line 734, in __protected_call__
    return self.run(*args, **kwargs)
  File "/src/iib/workers/tasks/utils.py", line 860, in wrapper
    return func(*args, **kwargs)
  File "/src/iib/workers/tasks/build.py", line 1095, in handle_rm_request
    merge_catalogs_dirs(catalog_from_db, catalog_from_index)
  File "/src/iib/workers/tasks/fbc_utils.py", line 86, in merge_catalogs_dirs
    raise IIBError(msg)
iib.exceptions.IIBError: config directory does not exist: /tmp/iib-u6rnxy3l/from_index/configs`

Could you please check?

File: iib/workers/tasks/build.py [hunk 1]
```diff
@@ -1077,7 +1077,35 @@ def handle_rm_request(
...
+            )
+
+            # generated by opm_migrate call inside opm_registry_rm_fbc
+            # rename `catalog` dir (because we need to use this name for
+            # final destination of catalog
+            catalog_from_db = os.path.join(temp_dir, 'from_db')
+            os.rename(os.path.join(temp_dir, 'catalog'), catalog_from_db)
+
+            # get catalog with opted-in operators
+            catalog_from_index = get_catalog_dir(
```
Comment (chandwanitulsi) [inline iib/workers/tasks/build.py:1090]: adding `os.makedirs(os.path.join(temp_dir, 'from_index'))` before this line does not give the error mentioned in PR

File: iib/workers/tasks/build.py [hunk 2]
```diff
@@ -1077,7 +1077,35 @@ def handle_rm_request(
                 operators=operators,
                 binary_image=prebuild_info['binary_image'],
                 overwrite_from_index_token=overwrite_from_index_token,
+                generate_cache=False,
+            )
+
+            # generated by opm_migrate call inside opm_registry_rm_fbc
+            # rename `catalog` dir (because we need to use this name for
+            # final destination of catalog
+            catalog_from_db = os.path.join(temp_dir, 'from_db')
```
[the remark this answers is not in the document]
Comment (lipoja) [reply iib/workers/tasks/build.py:1085]: updated - please check

**Review by chandwanitulsi — APPROVED** [review]
everything looks okay, it works after testing

**Review by yashvardhannanavati — APPROVED** [review]

**Comment by yashvardhannanavati** [comment]
@chandwanitulsi is following it up with the CI fix in a different PR

**Pull request merged**